lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -258,56 +258,43 @@ def add_merchant_description(post, options)
+        return unless options[:stored_credential]
 
-        options = stored_credential_options(stored_credential, options.fetch(:reason_code, ''))
-        post[:processingInformation][:commerceIndicator] = options.fetch(:transaction_type, 'internet')
-        stored_credential[:initial_transaction] ? initial_transaction(post, options) : subsequent_transaction(post, options)
+        post[:processingInformation][:commerceIndicator] = commerce_indicator(options.dig(:stored_credential, :reason_type))
+        add_authorization_options(post, options)
       end
 
-      def stored_credential_options(options, reason_code)
-        transaction_type = options[:reason_type]
-        transaction_type = 'install' if transaction_type == 'installment'
-        initiator = options[:initiator] if  options[:initiator]
-        initiator = 'customer' if initiator == 'cardholder'
-        stored_on_file = options[:reason_type] == 'recurring'
-        options.merge({
-          transaction_type: transaction_type,
-          initiator: initiator,
-          reason_code: reason_code,
-          stored_on_file: stored_on_file
-        })
+      def commerce_indicator(reason_type)
+        case reason_type
+        when 'recurring'
+          'recurring'
+        when 'installment'
+          'install'
+        else
+          'internet'
+        end
       end
 
-      def add_processing_information(initiator, merchant_initiated_transaction_hash = {})
-        {
+      def add_authorization_options(post, options)
+        initiator = options.dig(:stored_credential, :initiator) == 'cardholder' ? 'customer' : 'merchant'
+        authorization_options = {
           authorizationOptions: {
             initiator: {
-              type: initiator,
-              merchantInitiatedTransaction: merchant_initiated_transaction_hash,
-              storedCredentialUsed: true
+              type: initiator
             }
           }
         }.compact
-      end
 
-      def initial_transaction(post, options)
-        processing_information = add_processing_information(options[:initiator], {
-          reason: options[:reason_code]
-        })
-
-        post[:processingInformation].merge!(processing_information)
-      end
-
-      def subsequent_transaction(post, options)
-        network_transaction_id = options[:network_transaction_id] || options.dig(:stored_credential, :network_transaction_id) || ''
-        processing_information = add_processing_information(options[:initiator], {
-          originalAuthorizedAmount: post.dig(:orderInformation, :amountDetails, :totalAmount),
-          previousTransactionID: network_transaction_id,
-          reason: options[:reason_code],
-          storedCredentialUsed: options[:stored_on_file]
-        })
-        post[:processingInformation].merge!(processing_information)
+        authorization_options[:authorizationOptions][:initiator][:storedCredentialUsed] = true if initiator == 'merchant'
```

Comment:
Looking at their documentation this can be send for any subsequent transactions as long as it's not the initial transaction. 

> When a customer-initiated transaction (CIT) uses stored credentials, the transaction must include the following information in the authorization request:
• processingInformation.authorizationOptions.initiator.storedCredentialUsed–Set the field to true to indicate that the previously stored customer credentials were used.

Reply:
Where did you see that? Their API reference only mentions MITs for this

Reply:
Page 13 of https://developer.cybersource.com/content/dam/docs/cybs/en-us/payments/developer/ctv/rest/credentials.pdf